Базовая стилизация всех страниц

Kirill-hippo opened the pull request:
Внес на страницу index.html базовую стилизацию:
 - добавлен фон в хедер и футер
 - работа со шрифтами

---
:mortar_board: [Базовая стилизация всех страниц](https://up.htmlacademy.ru/htmlcss/35/user/2135219/tasks/8)

:boom: https://htmlacademy-htmlcss.github.io/2135219-device-35/4/

Kirill-hippo committed:
Базовая стилизация

Внес на страницу index.html базовую стилизацию:
 - добавлен фон в хедер и футер
 - работа со шрифтами

avilona commented on styles/styles.css:
```diff
@@ -0,0 +1,281 @@
+  line-height: 24px;
+  color: #000000;
+  background-color: #ffffff;
+}
```
почти у каждого html-элемента есть дефолтные браузерные стили, которые мы можем переопределять уже в наших стилях
в частности у body есть дефолтный margin, из-за чего получаются ненужные отступы по краям страницы:

![image](https://user-images.githubusercontent.com/12893725/200106608-b06901f6-db77-4ac4-a333-a32e4a84c75f.png)

в общем, этот дефолтный margin обязательно надо убрать

![image](https://user-images.githubusercontent.com/12893725/200106665-73cb48bd-7952-4227-86e4-321e2f3ea967.png)

avilona commented on styles/styles.css:
```diff
@@ -0,0 +1,281 @@
+  text-decoration: none;
+  letter-spacing: 0.2em;
+  color: inherit;
+}
```
для body ты уже прописал шрифт Raleway, таким образом он теперь наследуется большинством элементов автоматически, кроме button и input, но если у какой-то кнопки или инпута тоже должен быть шрифт Raleway, то хорошей практикой будет прописать его не напрямую, а так:

`font-family: inherit;`

это означает, что шрифт будет наследоваться от родительского элемента
на данный момент тяжело будет поддерживать проект, в котором для каждого класса напрямую прописан шрифт, если нужно будет менять основной шрифт на сайте, то это придётся сделать в куче мест, а также получается некоторое количество избыточного кода

avilona commented on styles/styles.css:
```diff
@@ -0,0 +1,281 @@
+  text-decoration: none;
+  letter-spacing: 0.2em;
+  color: inherit;
+}
```
![image](https://user-images.githubusercontent.com/12893725/200106936-aad7bb3c-7771-4abf-9280-33b8eef50da9.png)

текст Каталог товаров не по макету, там внутри избыточный тег \<strong>, который имеет дефолтный стиль `font-weight:bold (т.е. 700)`, что является приоритетней того что у тебя прописано для этого блока, т.е. `font-weight: 800;`

avilona commented on styles/styles.css:
```diff
@@ -0,0 +1,281 @@
+  font-weight: 400;
+  font-size: 18px;
+  line-height: 30px;
+  color: inherit;
```
в отличие от шрифта, цвет наоборот лучше не наследовать, т.е. цвет шрифта, заданный для родительского элемента опять же наследуется почти всеми дочерними элементами автоматически
ссылки, например, не наследуют и им нужно прописать цвет явно, а обычные текстовые элементы типа абзацев, списков и заголовков - наследуют и  у них у всех color: inherit нужно поубирать

avilona commented on styles/styles.css:
```diff
@@ -0,0 +1,281 @@
+  letter-spacing: 0.2em;
+  color: inherit;
+}
+body h2 {
+  font-family: 'Raleway', sans-serif;
+  font-weight: 800;
+  font-size: 50px;
+  line-height: 50px;
+  color: inherit;
+}
```
этими стилями ты глобально определил внешний вид всех заголовков h2 на сайте, так делать совсем нельзя, стилизовать надо по классам, либо по вложенности от класса, в критериях проверки проекта как раз есть такой [пункт](https://up.htmlacademy.ru/htmlcss/35/criteries#CSS-02)

avilona commented on styles/styles.css:
```diff
@@ -0,0 +1,281 @@
+  line-height: 30px;
+  color: #000000;
+}
+.other-products {
+  font-family: 'Raleway', sans-serif;
+  font-weight: 800;
+  font-size: 18px;
+  line-height: 24px;
+  letter-spacing: -0.02em;
+  text-decoration: none;
+}
```
![image](https://user-images.githubusercontent.com/12893725/200107514-660d0582-5276-4793-84ea-8a928b4cb8ed.png)

цвет шрифта не по макету, а также надо убрать избыточные теги \<p> внутри ссылок

avilona commented on styles/styles.css:
```diff
@@ -0,0 +1,281 @@
+  color: #444444;
+}
+
+.text-order-delivery{
+  font-family: 'Raleway', sans-serif;
+  width: 910px;
+  height: 20px;
+  font-weight: 800;
+  font-size: 20px;
+  line-height: 23px;
+  text-align: center;
+  letter-spacing: 0.2em;
+  text-transform: uppercase;
+  text-decoration: none;
+  color: #000000;
+}
```
![image](https://user-images.githubusercontent.com/12893725/200107566-a5282a8d-6969-40d9-a7cb-4225b3dd9b3b.png)

здесь достаточно оставить просто ссылку, а \<div>, в который она обернута убрать, а также текст не получается выровнять по центру потому, что у этого дива стоит ширина 125px, и здесь будет достаточно одного класса, по которому можно будет стилизовать весь элемент

avilona commented on index.html:
```diff
@@ -3,54 +3,55 @@
+          <li class="navigation-item">
+            <a class="catalog" href="#"><strong>Каталог товаров</strong></a>
+              <ul class="catalog-list" aria-label="Каталог товаров">
+                <li value="virtual-reality">Виртуальная реальность</li>
+                <li value="selfie-monopods">Моноподы для селфи</li>
+                <li value="action-camera">Экшн-камеры</li>
+                <li value="fitness-bracelets">фитнес-браслеты</li>
+                <li value="smart-watches">Умные часы</li>
+                <li value="quadracopters">Квадракоптеры</li>
```
атрибут value для \<li> не имеет никакого смысла, их надо поубирать

avilona commented on index.html:
```diff
@@ -66,16 +67,16 @@ <h1 class="visually-hidden">Интернет магазин гаджетов "Д
           <a class="more-detailed" href="#">Подробнее</a>
-          <p class="specifications-main"><span class="specifications">800 m</span> <br>Дальность полета</p>
-          <p class="specifications-main"><span class="specifications">500 m</span> <br>Радиус поражения</p>
+          <p class="specifications-main"><span class="specifications">800 м</span> <br>Дальность полета</p>
+          <p class="specifications-main"><span class="specifications">500 м</span> <br>Радиус поражения</p>
```
вместо тега \<br> лучше обернуть вторую часть текста в ещё один \<span>

avilona commented on index.html:
```diff
@@ -85,31 +86,31 @@ <h2>Делай селфи, как Бен Стиллер!</h2>
       <div class="other-products-list">
         <h2 class="visually-hidden">Другие товары</h2>
-          <a class="block-banner-virtual" href='#'>
+          <a class="other-products" href='#'>
             <p class="block-text">Виртуальная реальность</p>
           </a>
```
теги \<p> являются избыточными, здесь достаточно просто \<a>

avilona commented on index.html:
```diff
@@ -132,33 +133,33 @@ <h2 class="visually-hidden">Услуги</h2>
-          <li class="bout-us-item"><b>Сдик</b></li>
-          <li class="bout-us-item"><b>ТоксБери</b></li>
-        </ul>
+          <ul class="about-us-list">
+            <li class="bout-us-item"><b>Меловые линии</b></li>
+            <li class="bout-us-item"><b>Сдик</b></li>
+            <li class="bout-us-item"><b>ТоксБери</b></li>
+          </ul>
```
теги \<b> являются избыточными, их нужно поубирать, внешний вид шрифта сделать при помощи стилей

avilona commented on index.html:
```diff
@@ -132,33 +133,33 @@ <h2 class="visually-hidden">Услуги</h2>
       <p class="contacts">Вы можете забрать товар сами, заехав в наш офис. Заодно сможете проверить работоспособность покупки.</p>
-      <h4>Интернет-магазин"Девайс<h4>
+      <h4 class="heading-address">Интернет-магазин "Девайс"<h4>
       <p class='address'>Санкт-Петербург, Набережная реки Карповки, 5, литера П.</p>
```
везде используются двойные кавычки, а здесь - одинарные, это надо исправить
код должен быть написан единообразно, а также есть критерий про [кодгайд](https://up.htmlacademy.ru/htmlcss/35/criteries#PROJ-01)

Kirill-hippo committed:
Исправление ошибок ДЗ Базовой стилизации

Исправил ошибки базовой стилизации на странице index.html
Подготовил стилизацию для страницы catalog.html
